sktime/utils/_testing/estimator_checks.py

```diff
@@ -347,6 +349,12 @@ def check_raises_not_fitted_error(Estimator):
 def check_fit_idempotent(Estimator):
     # Check that calling fit twice is equivalent to calling it once
     estimator = _construct_instance(Estimator)
+
+    # Due to how clustering algorithms work, cant guarantee same
```

Comment:
Could you comment on this a bit more why we cannot handle this with random states?

Reply:
So with k-means and k-medoids how the clusters will form depend massivly on where the cluster centers initialise. As all of the cluster initilisation algorithms are based on probability it means we'll never have the same clusters being formed. 

I could create a new optional parameter for random state and use it for the initialisation of the clusters so we get conistent behaviour. In addition or as a alternate (although I was thinking of doing this at a later date) we could take 'starting_centers' as a optional parameter that will be used as the starting centers.

Reply:
if you could seed them that would be best, we do it with the classifiers, I think thats all thats needed at the moment

Reply:
Yes, passing a random state via `check_random_state` from scikit-learn would ensure reproducibility and is inline with the other algorithm implementations.